gnomad_qc/v4/create_release/update_hgdp_tgp_subset_AF.py

```diff
@@ -0,0 +1,484 @@
+
+    .. note::
+    This is the same as we did gnomad_qc/v3/create_release/create_release_sites_ht.py#L291-L310, except excluding the pops & downsampling options.
+
+    :param global_freq_ht: Global frequency HT.
+    :param subset_freq_hts: List of subset frequency HTs.
+    :return: Concatenated frequency HT.
+    """
+    freq_ht = hl.Table.multi_way_zip_join(
+        [global_freq_ht.select("freq").select_globals("freq_meta")] + subset_freq_hts,
+        data_field_name="freq",
+        global_field_name="freq_meta",
+    )
+    freq_ht = freq_ht.transmute(freq=freq_ht.freq.flatmap(lambda x: x.freq))
+    freq_ht = freq_ht.transmute_globals(
+        freq_meta=freq_ht.freq_meta.flatmap(lambda x: x.freq_meta)
+    )
+
+    # Create frequency index dictionary on concatenated array (i.e., including
+    # all subsets)
+    freq_ht = freq_ht.annotate_globals(
+        freq_index_dict=make_freq_index_dict(
+            freq_meta=hl.eval(freq_ht.freq_meta),
+        )
+    )
+    return freq_ht
+
+
+def main(args):
+    """Script to update update Allele Frequencies for HGDP + 1KG subset for v4."""
+    test = args.test
+
+    hl.init(
+        log="/update_AFs_HGDP+TGP.log",
+        default_reference="GRCh38",
+        tmp_dir="gs://gnomad-tmp-4day",
+    )
+
+    logger.info("Loading the HGDP + 1KG subset dense MT...")
+    mt = hl.read_matrix_table(hgdp_tgp_subset(dense=True))
+
+    logger.info("Loading the release HT...")
+    ht = hl.read_table(release_ht_path(release_version="3.1.2"))
+
+    logger.info("Updating the HGDP pop labels...")
+    ht = update_hgdp_pop_labels(ht, old_pop="miaozu", new_pop="miao")
+    ht = update_hgdp_pop_labels(ht, old_pop="yizu", new_pop="yi")
+    ht = update_hgdp_pop_labels(ht, old_pop="bantusafrica", new_pop="bantusouthafrica")
+
+    logger.info("Selecting `freq` and `freq_meta` from the release HT...")
+    ht = ht.select("freq").select_globals("freq_meta")
+
+    if test:
+        logger.info("Filtering to 10kb in DRD2 in MT for testing purposes...")
+        test_interval = [
+            hl.parse_locus_interval(
+                "chr11:113425000-113435000", reference_genome="GRCh38"
+            )
+        ]
+        mt = hl.filter_intervals(mt, test_interval)
+        ht = hl.filter_intervals(ht, test_interval)
+
+    logger.info("Loading HGDP_TGP subset meta HT...")
+    meta_ht = hgdp_tgp_subset_annotations(sample=True).ht()
+
+    logger.info("Adding updated sample QC annotations to meta HT...")
+    meta_ht = add_updated_sample_qc_annotations(meta_ht)
+    meta_ht = meta_ht.checkpoint(hgdp_tgp_meta_updated.path, _read_if_exists=True)
+
+    logger.info("Filtering samples in meta HT that will be added and subtracted...")
+    samples_to_add, samples_to_subtract = _get_filtered_samples(meta_ht)
+
+    logger.info("Annotating MT with the new meta HT...")
+    mt = mt.select_globals()
+    mt = mt.annotate_cols(**meta_ht[mt.col_key])
+
+    logger.info("Calculating AFs for added samples...")
```

Comment:
A lot of the code below is repeated, I think you can make it easier to read if you make it less redundant